Code/Library/WinAPI.cpp

```diff
@@ -610,6 +611,108 @@ wchar_t WinAPI::WideCharToUpper(wchar_t ch, int languageID)
 	}
 }
 
+std::wstring WinAPI::CharToWString(char c) {
+	WCHAR wideChar[2];
+	int wlen = MultiByteToWideChar(GetActiveKeyboardAnsiCP(), 0, &c, 1, wideChar, 2);
```

Comment:
Is it a good idea to query current keyboard layout and its code page for *every* input character?

By the way, believe it or not, input key events are dispatched in our code:

https://github.com/crymp-net/crymp-client/blob/83019074e2dfb8ee8290ad555a225cc807660ed6/Code/CrySystem/GameWindow.cpp#L56-L89

It's the reverse engineered original implementation. We can fix it and avoid this conversion back to wide char completely. The wide char is actually passed through `event.timestamp`. And `WM_INPUTLANGCHANGE` can be used to effectively detect keyboard layout changes without querying it.

What do you think?